src/storageService.ts

```diff
@@ -0,0 +1,44 @@
+import * as fs from "fs-extra";
+import * as path from "path";
+
+export class StorageService {
+  constructor(
+    public baseDirectory: string,
+  ) {}
+
+  public initSync(): void {
+    return fs.ensureDirSync(this.baseDirectory);
+  }
+
+  public getItemSync<T>(itemName: string): T | null {
+    try {
+      const filePath = path.resolve(this.baseDirectory, itemName);
+      return fs.readJsonSync(filePath);
+    } catch (e) {
+      return null;
```

Comment:
AFAIK getItemSync is currently used in `loadCachedPlatformAccessoriesFromDisk()`.
What I currently don't like that errors in the StorageService are complexity silenced.

Additionally `loadCachedPlatformAccessoriesFromDisk()` will just do nothing if getItemSync returns `null`.

Am I missing anything that this "empty" catch makes sense?
If we encounter an IO error, we would first of all **never** the error message, and homebridge would just not load cached accessories silently.

I would argue for a solution where homebridge fails to init if the file can't be loaded (maybe including some logic to repair permissions, which are no 1. issue I think).


Same applies to the Promise based method.

Reply:
The main reasons this would throw would be if the file does not exist yet or if the file is corrupted (not valid json) - in either case we would want to write a new cache file.

We could check for the files existence before attempting to read it so we can log an error if the file is corrupted or we don't have permissions to read it - that just requires one more stat call.

I wouldn't prevent the bridge from starting if the JSON isn't valid - just re-write and move on.

Reply:
Ah. I would suggest to make that explicit then. So comment those individual scenarios in the catch clause (maybe adding a debug log statement of the logger?).

And for the corrupted file scenario, I would propose that we try as much as possible to recover (e.g. ensure permissions set) and outer wise overwrite as you suggest, though with logging a warning. So we can easily debug why cachedAccessories was reset.

Reply:
Changes have been made - red error log messages will explain that the cached accessories either cannot be saved or cannot be loaded and the implications of this. It will not prevent Homebridge from loading.

Also updated the `loadCachedPlatformAccessoriesFromDisk` to use the async methods, as the only places we use this is in async functions, and async is better so why not 😄